### ISSUE TITLE
Provide a generic image

### ISSUE THREAD

**Issue by ufoscout** [testcontainers/testcontainers-rs#56]
It would be good to have an equivalent of the [GenericContainer](https://www.testcontainers.org/features/creating_container/) Java class to easily use custom images.

In fact, at the moment, for using an image not already provided by testcontainers-rs, it is required to create a new implementation of the [Image](https://github.com/testcontainers/testcontainers-rs/blob/master/core/src/image.rs) trait which involves lots of boilerplate.

Is there any plan to add it?

**Comment by thomaseizinger** [comment]
> Is there any plan to add it?

At the moment no because we don't have a need for it right now.

> It would be good to have an equivalent of the GenericContainer Java class to easily use custom images.

Would containers created in this way have any "wait strategy" attached to them? How do you know when they are ready to be interacted with? From the docs of testcontainers-java, I assume this feature is then not available?

I guess we could start out with a simple generic container and add parametrized wait strategies later on.

**Comment by ufoscout** [comment]
> Would containers created in this way have any "wait strategy" attached to them? How do you know when they are ready to be interacted with? From the docs of testcontainers-java, I assume this feature is then not available?

Yes, they have a DSL that permits to define everything, including a wait strategy. For example, this is the way I create a custom Postgres image:
```java
public GenericContainer containerWithWaitStrategy = new GenericContainer("postgres:9.6-alpine")
                    .withEnv("POSTGRES_PASSWORD", "postgres")
                    .withEnv("POSTGRES_USER", "postgres")
                    .withExposedPorts(5432)
                    .waitingFor(
                            Wait.forListeningPort()
                           // OR: Wait.forLogMessage(".*ready to accept connections*\\s", 1)
                    );
```

**Comment by D4nte** [comment]
Closed in #58
